### PR TITLE
Update GET /eholdings/proxy-types to support multiple KB credentials

### PR DESCRIPTION
**Pull request by natali-zaitseva**
## Purpose
Implementation story for https://issues.folio.org/browse/MODKBEKBJ-416

## Approach
* modified existing proxyTypes.json file to include credentialsId and total results parameters.
* added GET /eholdings/kb-credentials/<credentialsId>/proxy-types
* added tests for GET /eholdings/kb-credentials/<credentialsId>/proxy-types
* modified the logic for GET /eholdings/proxy-types
* updated tests for GET /eholdings/proxy-types

### CODE REVIEW

File: descriptors/ModuleDescriptor-template.json [hunk 1]
```diff
@@ -633,6 +643,7 @@
         "kb-ebsco.resources.item.put",
         "kb-ebsco.resources.item.delete",
         "kb-ebsco.resource-tags.put",
+        "kb-ebsco.kb-credentials.proxy-types.collection.get",
```
Comment (psmagin) [inline descriptors/ModuleDescriptor-template.json:646]: This permission needs to be sub permission of `kb-ebsco.kb-credentials.all`

File: ramls/proxies.raml [hunk 2]
```diff
@@ -30,6 +30,24 @@ types:
...
+/eholdings/kb-credentials/{id}/proxy-types:
+  displayName: Proxy Types
+  description: List of supported proxy types.
+  uriParameters:
+    id:
+      pattern : "^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[1-5][0-9a-fA-F]{3}-[89abAB][0-9a-fA-F]{3}-[0-9a-fA-F]{12}$"
+  get:
+    description: Get a list of supported proxy types for KB Credentials.
+    responses:
+      200:
```
Comment (psmagin) [inline ramls/proxies.raml:42]: 404 if credentials are missing also a probable response

File: src/main/java/org/folio/service/proxytypes/ProxyTypesServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  @Override
+  public CompletableFuture<ProxyTypes> findByUser(Map<String, String> okapiHeaders) {
+    return credentialsService.findByUser(okapiHeaders)
+      .thenCompose(this::retrieveProxies);
+  }
+
+  private CompletableFuture<ProxyTypes> retrieveProxies(KbCredentials kbCredentials) {
+    final HoldingsIQServiceImpl holdingsIQService = createHoldingsIQService(kbCredentials);
+    return holdingsIQService.retrieveProxies()
+      .thenApply(proxyTypesConverter:: convert)
```
Comment (psmagin) [inline src/main/java/org/folio/service/proxytypes/ProxyTypesServiceImpl.java:51]:
```suggestion
      .thenApply(proxyTypesConverter::convert)
```
extra space

File: src/test/java/org/folio/rest/impl/integrationsuite/EHoldingsProxyTypesImplTest.java [hunk 4]
```diff
@@ -1,74 +1,188 @@
...
   private static final String RMI_PROXIES_URL = "/rm/rmaccounts/" + STUB_CUSTOMER_ID + "/proxies.*";
   private static final String EHOLDINGS_PROXY_TYPES_URL = "eholdings/proxy-types";
+  private static final String EHOLDINGS_PROXY_TYPES_BY_CREDENTIALS_ID_URL = "/eholdings/kb-credentials/%s/proxy-types";
+  private static final String STUB_CREDENTILS_ID = "12312312-1231-1231-a111-111111111111";
+  private static final String JOHN_ID = "47d9ca93-9c82-4d6a-8d7f-7a73963086b9";
+  private static final String JANE_ID = "781fce7d-5cf5-490d-ad89-a3d192eb526c";
+  private static String johnToken = "eyJhbGciOiJIUzI1NiJ9." +
+    "eyJzdWIiOiJqb2huX2RvZSIsInVzZXJfaWQiOiI0N2Q5Y2E5My05YzgyLTRkNmEtOGQ3Zi03YTczOTYzMDg2Y" +
+    "jkiLCJpYXQiOjE1ODU4OTUxNDQsInRlbmFudCI6ImZzIn0.HTx-4aUFIPtEHO-6ZcYML6K3-0VRDGv3KX44JoT3hxg";
+  private static String janeToken = "eyJhbGciOiJIUzI1NiJ9." +
```
Comment (psmagin) [inline src/test/java/org/folio/rest/impl/integrationsuite/EHoldingsProxyTypesImplTest.java:47]: also can be constants

File: src/main/java/org/folio/rest/impl/EholdingsProxyTypesImpl.java [hunk 5]
```diff
@@ -11,26 +14,48 @@
...
+  public void getEholdingsProxyTypes(Map<String, String> okapiHeaders,
+                                     Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+
+    proxyTypesService.findByUser(okapiHeaders)
+      .thenAccept(proxyTypes -> asyncResultHandler.handle(succeededFuture(
+        GetEholdingsProxyTypesResponse.respond200WithApplicationVndApiJson(proxyTypes))))
+      .exceptionally(handleException(asyncResultHandler));
+  }
+
+  @Override
```
Comment (psmagin) [inline src/main/java/org/folio/rest/impl/EholdingsProxyTypesImpl.java:45]:
```suggestion
  @Override @Validate @HandleValidationErrors
```

Comment (natali-zaitseva) [reply src/main/java/org/folio/rest/impl/EholdingsProxyTypesImpl.java:46]: annotations added

File: ramls/proxies.raml (hunk 2 above)
Comment (natali-zaitseva) [reply ramls/proxies.raml:42]: added

**Pull request merged**